TypeCobol/Compiler/Diagnostics/NodeCheckers.cs

```diff
@@ -0,0 +1,32 @@
+﻿using TypeCobol.Compiler.CodeElements;
+using TypeCobol.Compiler.Nodes;
+using TypeCobol.Compiler.Parser;
+
+namespace TypeCobol.Compiler.Diagnostics
+{
+    #region NodeCheckers
+
+    class WhenConditionNodeChecker
+    {
+        public static void OnNode(Node node)
```

Comment:
Argument could be more strictly typed as `When`

Reply:
In fact at parsing time it should be a Then node, but the parser is not aware about that can only assert here